uni/lib/view/Pages/course_units_page_view.dart

```diff
@@ -0,0 +1,190 @@
+          const PageTitle(name: constants.navCourseUnits),
+          const Spacer(),
+          DropdownButton<String>(
+            disabledHint: const Text('Ano'),
+            value: selectedSchoolYear,
+            icon: const Icon(Icons.arrow_drop_down),
+            onChanged: (String? newValue) {
+              setState(() => selectedSchoolYear = newValue!);
+            },
+            items: availableYears.map<DropdownMenuItem<String>>((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Container(
+                    padding: const EdgeInsets.symmetric(horizontal: 5),
+                    child: Text(value)),
+              );
+            }).toList(),
+          ),
+          const SizedBox(width: 10),
+          DropdownButton<String>(
+            disabledHint: const Text('Semestre'),
+            value: selectedSemester,
+            icon: const Icon(Icons.arrow_drop_down),
+            onChanged: (String? newValue) {
+              setState(() => selectedSemester = newValue!);
+            },
+            items: availableSemesters
+                .map<DropdownMenuItem<String>>((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Container(
+                    padding: const EdgeInsets.symmetric(horizontal: 5),
+                    child: Text(value)),
+              );
+            }).toList(),
+          ),
+          const SizedBox(width: 20)
+        ],
+      ),
+      RequestDependentWidgetBuilder(
+          context: context,
+          status: requestStatus ?? RequestStatus.none,
+          contentGenerator: generateCourseUnitsCards,
+          content: filteredCourseUnits ?? [],
+          contentChecker: courseUnits?.isNotEmpty ?? false,
+          onNullContent: Center(
+            heightFactor: 10,
+            child: Text('Não existem cadeiras para apresentar',
+                style: Theme.of(context).textTheme.headline6),
+          ))
+    ]);
+  }
+
+  Widget generateCourseUnitsCards(courseUnits, context) {
+    if ((courseUnits as List<CourseUnit>).isEmpty) {
+      return Center(
+          heightFactor: 10,
+          child: Text('Sem cadeiras no período selecionado',
+              style: Theme.of(context).textTheme.headline6));
+    }
+
+    List<Widget> rows = [];
+    for (var i = 0; i < courseUnits.length; i += 2) {
+      if (i < courseUnits.length - 1) {
+        rows.add(IntrinsicHeight(
+            child:
+                Row(crossAxisAlignment: CrossAxisAlignment.stretch, children: [
+          Flexible(child: CourseUnitCard(courseUnits[i])),
+          const SizedBox(width: 10),
+          Flexible(child: CourseUnitCard(courseUnits[i + 1])),
+        ])));
+      } else {
+        rows.add(Row(children: [
+          Flexible(child: CourseUnitCard(courseUnits[i])),
+          const SizedBox(width: 10),
+          const Spacer()
+        ]));
+      }
+    }
```

Comment:
Could we use a GridView to display this? It was confusing at first to know what this does

Reply:
No. GridView forces all rows to have the same height (aspect ratio). This manual way of doing things allows variable row height and card stretch next to its row neighbor and thus a better look.
I'll extract this for readability